includes/class-amp-theme-support.php

```diff
@@ -177,6 +177,7 @@ public static function register_hooks() {
 		add_action( 'wp_head', array( __CLASS__, 'add_amp_custom_style_placeholder' ), 8 ); // Because wp_print_styles() normally happens at 8.
 		add_action( 'wp_head', array( __CLASS__, 'add_amp_component_scripts' ), 10 );
 		add_action( 'wp_head', 'amp_add_generator_metadata', 20 );
+		add_filter( 'wp_video_shortcode_override', array( __CLASS__, 'video_override' ), 10, 2 );
```

Comment:
I'm thinking that `video_override` actually would be better put into either a new video embed class, or else to add the method to the Vimeo and YouTube. This would ensure that Vimeo and YouTube videos used with the `video` shortcode will work in the old paired mode templates as well. It shouldn't be limited to theme support.